### PR TITLE
Processing chain validator

### PR DESCRIPTION
**Pull request by SteRiccio**

### CODE REVIEW

File: webapp/loggedin/modules/analysis/processingChain/actions.js [hunk 1]
```diff
@@ -102,43 +115,72 @@ export const saveProcessingChain = () => async (dispatch, getState) => {
...
   )(state)
+
+  const stepValidation = step ? await ProcessingChainValidator.validateStep(step) : null
+
   const calculation = R.pipe(
     ProcessingStepCalculationState.getCalculation,
     ProcessingStepCalculation.dissocTemporary,
     R.when(R.isEmpty, R.always(null)),
   )(state)
+  const calculationValidation = await ProcessingChainValidator.validateCalculation(calculation, surveyDefaultLang)
```
Comment (minotogna) [inline webapp/loggedin/modules/analysis/processingChain/actions.js]: what is calculation is null? (you added the check to stepValidation)

File: webapp/loggedin/modules/analysis/processingStepCalculation/actions.js [hunk 2]
```diff
@@ -3,33 +3,44 @@ import * as R from 'ramda'
...
 import * as SurveyState from '@webapp/survey/surveyState'
+import * as ProcessingChainState from '@webapp/loggedin/modules/analysis/processingChain/processingChainState'
 import * as ProcessingStepState from '@webapp/loggedin/modules/analysis/processingStep/processingStepState'
 import * as ProcessingStepCalculationState from './processingStepCalculationState'
 
 import { showAppLoader, hideAppLoader } from '@webapp/app/actions'
 import { showNotification } from '@webapp/app/appNotification/actions'
 import { nodeDefCreate } from '@webapp/survey/nodeDefs/actions'
 import { navigateToNodeDefEdit } from '@webapp/loggedin/modules/analysis/actions'
+import { processingChainValidationUpdate } from '../processingChain/actions'
```
Comment (minotogna) [inline webapp/loggedin/modules/analysis/processingStepCalculation/actions.js]: I would use ```@webapp...```

Comment (SteRiccio) [reply webapp/loggedin/modules/analysis/processingStepCalculation/actions.js]: ok

File: webapp/loggedin/modules/analysis/processingChain/actions.js (hunk 1 above)
Comment (SteRiccio) [reply webapp/loggedin/modules/analysis/processingChain/actions.js]: right, I'll check it

**Pull request merged**